objects/screen.c

```diff
@@ -241,6 +243,143 @@ screens_exist(void)
     return globalconf.screens.len > 0;
 }
 
+#ifdef XCB_RANDR_GET_MONITORS
+static bool
+screen_scan_randr_monitors(void)
+{
+    xcb_randr_get_monitors_cookie_t monitors_c = xcb_randr_get_monitors(globalconf.connection, globalconf.screen->root, 1);
+    xcb_randr_get_monitors_reply_t *monitors_r = xcb_randr_get_monitors_reply(globalconf.connection, monitors_c, NULL);
+    xcb_randr_monitor_info_iterator_t monitor_iter;
+    bool found = false;
+
+    for(monitor_iter = xcb_randr_get_monitors_monitors_iterator(monitors_r); monitor_iter.rem; xcb_randr_monitor_info_next(&monitor_iter))
+    {
+        lua_State *L = globalconf_get_lua_State();
+        screen_t *new_screen;
+        screen_output_t output;
+        xcb_randr_output_t *randr_outputs;
+        xcb_get_atom_name_cookie_t name_c;
+        xcb_get_atom_name_reply_t *name_r;
+
+        if(!xcb_randr_monitor_info_outputs_length(monitor_iter.data))
+            continue;
+
+        new_screen = screen_new(L);
+        new_screen->geometry.x = monitor_iter.data->x;
+        new_screen->geometry.y = monitor_iter.data->y;
+        new_screen->geometry.width= monitor_iter.data->width;
+        new_screen->geometry.height= monitor_iter.data->height;
+
+        output.mm_width = monitor_iter.data->width_in_millimeters;
+        output.mm_height = monitor_iter.data->height_in_millimeters;
+
+        name_c = xcb_get_atom_name_unchecked(globalconf.connection, monitor_iter.data->name);
+        name_r = xcb_get_atom_name_reply(globalconf.connection, name_c, NULL);
+        if (name_r) {
+            const char *name = xcb_get_atom_name_name(name_r);
+            size_t len = xcb_get_atom_name_name_length(name_r);
+
+            output.name = memcpy(p_new(char *, len + 1), name, len);
+            output.name[len] = '\0';
+            p_delete(&name_r);
+        } else {
+            output.name = strdup("unknown");
+        }
+        randr_output_array_init(&output.outputs);
+
+        randr_outputs = xcb_randr_monitor_info_outputs(monitor_iter.data);
+        for(int i = 0; i < xcb_randr_monitor_info_outputs_length(monitor_iter.data); i++) {
+            randr_output_array_append(&output.outputs, randr_outputs[i]);
+        }
+
+        screen_output_array_append(&new_screen->outputs, output);
```

Comment:
I can see why you don't want the deduplication that `screen_add` does, but due to this you also skip emission of the `added` signal. That will have to be fixed somehow. I suggest renaming `screen_add` to `screen_add_deduplicate` (feel free to come up with a better name, I don't like this one) and then having a new `screen_add` that just handles the "signal stuff".